src/components/eventListeners.js

```diff
@@ -1,68 +1,133 @@
+
+function onMyLocationButtonClick() {
   // get random location from suggestion list
-  state.currentSearchItem =
-    state.suggestions[Math.floor(Math.random() * state.suggestions.length)];
-  SEARCH_TEXT_INPUT.value = `${state.currentSearchItem.area_type}, ${state.currentSearchItem.city}, ${state.currentSearchItem.state_code}`;
+  state.currentSearchItem = state.suggestions[Math.floor(Math.random() * state.suggestions.length)];
+  searchTextInput.value = `${state.currentSearchItem.area_type}, ${state.currentSearchItem.city}, ${state.currentSearchItem.state_code}`;
 }
+
 function onGoButtonClick() {
-  // getProperties();
+  if (!state.currentSearchItem.city) {
+    searchTextInput.focus();
+    return;
+  }
+
   state.recentSearches.unshift(state.currentSearchItem);
+
   while (state.recentSearches.length > 5) {
     state.recentSearches.pop();
   }
+
   localStorage.setItem('recentSearches', JSON.stringify(state.recentSearches));
-  SEARCH_TEXT_INPUT.value = '';
+  searchTextInput.value = '';
   renderSearchList();
+  state.currentSearchItem = {};
 }
 
-function onRecentSearchClick(e) {
-  if (e.target.nodeName === 'SPAN') {
-    const index = Array.from(RECENT_SEARCHES_LIST.children).findIndex(
+async function onRecentSearchClick(e) {
+  const spinner = new Spinner(propertySpinnerContainer, propertySpinnerCanvas);
+
+  try {
+    if (e.target.nodeName !== 'SPAN') return;
+
+    state.propertyOffset = 0;
+    const index = Array.from(recentSearchList.children).findIndex(
       item => item.outerText === e.target.innerText
     );
     state.currentSearchListItem = state.recentSearches[index];
-    console.log(state.currentSearchListItem);
+
+    while (propertyList.firstChild) {
+      propertyList.removeChild(propertyList.firstChild);
+    }
+
+    propertyContainer.style.display = 'flex';
+    searchContainer.style.display = 'none';
+
+    spinner.toogleVisibility(true);
+    await getProperties();
+    renderProperties();
+  } catch (err) {
+    console.log(err);
+  } finally {
+    spinner.toogleVisibility(false);
   }
 }
 
+function onLoadMoreButtonClick() {
+  state.propertyOffset = state.propertyOffset + 20;
+  renderProperties();
+}
+
+function onBackButtonClick() {
+  propertyContainer.style.display = 'none';
+  searchContainer.style.display = 'flex';
+}
+
 export function initEventListeners() {
-  SEARCH_TEXT_INPUT.addEventListener('focus', onFocus, true);
-  SEARCH_TEXT_INPUT.addEventListener('blur', onBlur, true);
-  SEARCH_TEXT_INPUT.addEventListener('input', onInput);
-  SUGGESTION_LIST.addEventListener('click', onSuggestionClick);
-  MY_LOCATION_BUTTON.addEventListener('click', onMyLocationClick);
-  GO_BUTTON.addEventListener('click', onGoButtonClick);
-  RECENT_SEARCHES_LIST.addEventListener('click', onRecentSearchClick);
+  searchTextInput.addEventListener('focus', onSearchFocus, true);
+  searchTextInput.addEventListener('blur', onSearchBlur, true);
+  searchTextInput.addEventListener('input', onSearchInput);
```

Comment:
I guess you don't need anonymous self-invoke function, just write:
searchTextInput.addEventListener('focus', onSearchFocus(), true)
, and remove anonymous self-invoke function